lib/discovery_model/feature_extractor.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2016, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE.md file.
+
+library smart.discovery_model.feature_extractor;
+
+import 'package:analyzer/src/generated/ast.dart';
+import 'package:analyzer/src/generated/java_io.dart';
+import 'package:sintr_common/logging_utils.dart' as log;
+
+import '../analysis_utils/analysis_utils.dart' as analysis_utils;
+
+import '../analysis_utils/type_utils.dart';
+
+// TODO(luekchurch): Refactor this so it shares an implementation with the
+// completion_perf driver
+
+class Analysis {
+  Stopwatch sw;
+  Analysis(String sdkPath) {
+    JavaSystemIO.setProperty("com.google.dart.sdk", sdkPath);
+  }
+
+  List<Map> analyzeSpecificFile(String path) {
+    log.trace("analyzeSpecificFile: $path");
+    sw = new Stopwatch()..start();
+
+    CompilationUnit compilationUnit = analysis_utils.setupAnalysis(path, sw);
```

Comment:
If you are analyzing all the files in a folder, then they probably can all live in the same analysis context. Setting up a new context per file can be expensive. I don't know if that's an issue here, but something to think about.